When working with a component that contains multiple independent units,
KiCad warns about pin locations being occupied even if the conflicting
pin is in a different unit.

This may to be unnecessary. This patch suppresses the warning if the
pins are in different units.

- Werner

Index: kicad/eeschema/pinedit.cpp
===================================================================
--- kicad.orig/eeschema/pinedit.cpp	2009-05-20 05:21:13.000000000 -0300
+++ kicad/eeschema/pinedit.cpp	2009-05-20 05:35:53.000000000 -0300
@@ -193,6 +193,9 @@
             continue;
         if( newpos != Pin->m_Pos )
             continue;
+        if( Pin->m_Unit != CurrentPin->m_Unit &&
+	    Pin->m_Unit && CurrentPin->m_Unit )
+            continue;
         if( Pin->m_Flags )
             continue;
         if( ask_for_pin )
